SDK/AppCenter/Microsoft.AppCenter.Shared/AppCenter.cs

```diff
@@ -11,51 +11,89 @@ namespace Microsoft.AppCenter
     /// </summary>
     public partial class AppCenter
     {
-        // Gets the first instance of an app secret corresponding to the given platform name, or returns the string 
+        const char Delimiter = ';';
+        const string PlatformIndicator = "=";
+        const string TargetPostfix = "Target";
+        const string SecretPostfix = "appSecret";
+
+        // Gets the first instance of an app sceret and/or target token corresponding to the given platform name, or returns the string 
         // as-is if no identifier can be found. Logs a message if no identifiers can be found.
-        internal static string GetSecretForPlatform(string secrets, string platformIdentifier)
+        internal static string GetSecretAndTargetForPlatform(string secrets, string platformIdentifier)
         {
+            var platformTargetIdentifier = platformIdentifier + TargetPostfix;
             if (string.IsNullOrEmpty(secrets))
             {
                 throw new AppCenterException("App secrets string is null or empty");
             }
 
             // If there are no equals signs, then there are no named identifiers, but log a message in case the developer made 
             // a typing error.
-            if (!secrets.Contains("="))
+            if (!secrets.Contains(PlatformIndicator))
             {
                 AppCenterLog.Debug(AppCenterLog.LogTag, "No named identifier found in appSecret; using as-is");
                 return secrets;
             }
 
             var parseErrorMessage = $"Error parsing key for '{platformIdentifier}'";
 
-            var platformIndicator = platformIdentifier + "=";
+            var platformIndicator = platformIdentifier + PlatformIndicator;
+            var platformTargetIdicator = platformTargetIdentifier + PlatformIndicator;
             var secretIdx = secrets.IndexOf(platformIndicator, StringComparison.Ordinal);
-            if (secretIdx == -1)
+            var targetTokenIdx = secrets.IndexOf(platformTargetIdicator, StringComparison.Ordinal);
+            var targetIdx = secrets.IndexOf(TargetPostfix.ToLower(), StringComparison.Ordinal);
+            if (secretIdx == -1 && targetTokenIdx == -1 && targetIdx == -1)
+            {
+                throw new AppCenterException(parseErrorMessage);
+            }
+            if (targetIdx >= 0 && secretIdx == -1 && targetTokenIdx == -1) 
+            {
+                AppCenterLog.Debug(AppCenterLog.LogTag, "Found named identifier 'target' in the secret. Returning as-is.");
+                return secrets;
+            }
+            if (secretIdx >= 0)
```

Comment:
This code is too complex for what it's doing and hard to read.
Let's use Split and Linq, example: https://stackoverflow.com/a/11673966

Reply:
To combine with https://stackoverflow.com/a/3319086 in case there are duplicate keys.